tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/microprofiile/template/ConfigPropertyTestTemplate.java

```diff
@@ -0,0 +1,162 @@
+import org.eclipse.reddeer.eclipse.condition.ProjectExists;
+import org.eclipse.reddeer.eclipse.jdt.ui.packageview.PackageExplorerPart;
+import org.eclipse.reddeer.eclipse.ui.views.markers.ProblemsView;
+import org.eclipse.reddeer.eclipse.ui.views.markers.ProblemsView.ProblemType;
+import org.eclipse.reddeer.swt.condition.ControlIsEnabled;
+import org.eclipse.reddeer.swt.condition.ShellIsAvailable;
+import org.eclipse.reddeer.swt.impl.button.PushButton;
+import org.eclipse.reddeer.swt.impl.combo.LabeledCombo;
+import org.eclipse.reddeer.swt.impl.ctab.DefaultCTabItem;
+import org.eclipse.reddeer.swt.impl.group.DefaultGroup;
+import org.eclipse.reddeer.swt.impl.menu.ContextMenuItem;
+import org.eclipse.reddeer.swt.impl.menu.ShellMenuItem;
+import org.eclipse.reddeer.swt.impl.shell.DefaultShell;
+import org.eclipse.reddeer.swt.impl.text.LabeledText;
+import org.eclipse.reddeer.swt.impl.tree.DefaultTreeItem;
+import org.eclipse.reddeer.workbench.condition.EditorHasValidationMarkers;
+import org.eclipse.reddeer.workbench.core.condition.JobIsRunning;
+import org.eclipse.reddeer.workbench.impl.editor.DefaultEditor;
+import org.eclipse.reddeer.workbench.impl.editor.TextEditor;
+import org.jboss.tools.cdi.bot.test.CDITestBase;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * 
+ * @author zcervink@redhat.com
+ * 
+ */
+public class ConfigPropertyTestTemplate extends CDITestBase {
+
+	@Before
+	public void prepareProject() {
+		PackageExplorerPart pexplorer = new PackageExplorerPart();
+		pexplorer.open();
+		pexplorer.getProject(PROJECT_NAME).select();
+
+		beansHelper.createClass(CDI_BEAN_1_JAVA_FILE_NAME, PACKAGE_NAME);
+		editResourceUtil.replaceClassContentByResource(CDI_BEAN_1_JAVA_FILE_NAME + JAVA_FILE_EXTENSION,
+				readFile("resources/configProperty/CdiBean1.jav_"), false, false);
+		TextEditor te = new TextEditor(CDI_BEAN_1_JAVA_FILE_NAME + JAVA_FILE_EXTENSION);
+		new WaitWhile(new EditorHasValidationMarkers(te));
+		te.save();
+		new WaitWhile(new EditorHasValidationMarkers(te));
+
+		convertToMavenProject(PROJECT_NAME, "war", false);
+		addDependency(PROJECT_NAME, "org.eclipse.microprofile.config", "microprofile-config-api", "2.0");
+		createApplicationPropertiesFile(PROJECT_NAME);
+	}
+
+	@Test
+	public void isConfigPropertyValidationAllowedTest() {
+		PackageExplorerPart pexplorer = new PackageExplorerPart();
+		pexplorer.open();
+		pexplorer.getProject(PROJECT_NAME)
+				.getProjectItem("src/main/java", PACKAGE_NAME, CDI_BEAN_1_JAVA_FILE_NAME + JAVA_FILE_EXTENSION).open();
+
+		ProblemsView pw = new ProblemsView();
+		pw.open();
+		int error_count = pw.getProblems(ProblemType.ERROR).size();
+		assertEquals("There are errors after including the Microprofile @ConfigProperty annotation into the project.",
+				error_count, 0);
+	}
+
+	private void convertToMavenProject(String projectName, String defaultPackaging, boolean withDependencies) {
+		PackageExplorerPart pexplorer = new PackageExplorerPart();
```

Comment:
Is not this code reused from another test-framework plugin? 
question here is if it is necessary.

Reply:
Firstly I was afraid of those dependencies between test-frameworks. Since I have found it is in the same repo I must say this duplicit method should be removed. Fixed, removed.

Reply:
thanks.